src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -149,6 +201,8 @@ public void removeTaskId(String taskId) {
         if (entry.getValue() != null && entry.getValue().equals(taskId)) {
           journalNodes.put(entry.getKey(), null);
           setJournalNodes(journalNodes);
+          Date date = DateUtils.addSeconds(new Date(), conf.getDeadNodeTimeout());
+          deadJournalNodeTimeStamp = new Timestamp(date.getTime());
```

Comment:
I'll also point out again that deadDNs shouldn't stop us from starting more DNs on new hosts. If you remove that constraint, then I would agree that JNs/NNs repeatedly failing is very unlikely, given that there are at most 3 of them.